app/services/irs_attempts_api/redis_client.rb

```diff
@@ -0,0 +1,44 @@
+module IrsAttemptsApi
+  class RedisClient
+    cattr_accessor :redis_pool do
+      ConnectionPool.new(size: 10) do
+        Redis::Namespace.new(
+          'irs-attempt-api',
+          redis: Redis.new(url: IdentityConfig.store.redis_irs_attempt_api_url),
+        )
+      end
+    end
+
+    def write_event(event_data)
+      key = event_data.jti
+      redis_pool.with do |client|
+        client.set(key, event_data.to_json)
+        client.expire(key, IdentityConfig.store.irs_attempt_api_event_ttl_seconds)
+      end
+    end
+
+    def delete_events(event_ids)
+      redis_pool.with do |client|
+        client.del(*event_ids)
+      end
+    end
+
+    def read_events(count = 1000)
+      redis_pool.with do |client|
+        keys = client.scan(0, count: count).last
+        client.mapped_mget(*keys).transform_values do |stringified_event_data|
+          next unless stringified_event_data.present?
+          event_data = JSON.parse(stringified_event_data)
+          IrsAttemptsApi::Event.new(event_data)
```

Comment:
maybe we could add an `IrsAttemptsApi::Event.from_json` and simplify this part?
```suggestion
          IrsAttemptsApi::Event.from_json(stringified_event_data)
```

Reply:
I actually thought `#from_json` was a method on `Struct` and was sad to learn that I was mistaken.